src/FilesSidebarTabApp.vue

```diff
@@ -19,26 +19,294 @@
 </template>
 
 <script>
+// Explicitly enable smart-tabs so it is also applied to the CSS rules.
+/* eslint no-mixed-spaces-and-tabs: ["error", "smart-tabs"] */
+
+import { getFileConversation } from './services/filesIntegrationServices'
+import { joinConversation, leaveConversation } from './services/participantsService'
+import CancelableRequest from './utils/cancelableRequest'
+import { getCurrentUser } from '@nextcloud/auth'
+import Axios from '@nextcloud/axios'
+import ChatView from './components/ChatView'
+
 export default {
+
 	name: 'FilesSidebarTabApp',
+
+	components: {
+		ChatView,
+	},
+
 	data() {
 		return {
 			// needed for reactivity
 			Talk: OCA.Talk,
+			sidebarState: OCA.Files.Sidebar.state,
+			/**
+			 * Stores the cancel function returned by `cancelableLookForNewMessages`,
+			 */
+			cancelGetFileConversation: () => {},
+			isTalkSidebarSupportedForFile: undefined,
 		}
 	},
+
 	computed: {
 		fileInfo() {
-			return this.Talk.fileInfo
+			return this.Talk.fileInfo || {}
+		},
+		fileId() {
+			return this.fileInfo.id
+		},
+		token() {
+			return this.$store.getters.getToken()
+		},
+		fileIdForToken() {
+			return this.$store.getters.getFileIdForToken()
+		},
+		isChatTheActiveTab() {
+			// FIXME check for empty active tab is currently needed because the
+			// activeTab is not set when opening the sidebar from the "Details"
+			// action (which opens the first tab, which is the Chat tab).
+			return !this.sidebarState.activeTab || this.sidebarState.activeTab === 'chat'
+		},
+	},
+
+	watch: {
+		fileInfo: {
+			immediate: true,
+			handler(fileInfo) {
+				if (this.token && (!fileInfo || fileInfo.id !== this.fileIdForToken)) {
+					this.leaveConversation()
+				}
+
+				this.setTalkSidebarSupportedForFile(fileInfo)
+			},
+		},
+
+		isChatTheActiveTab: {
+			immediate: true,
+			handler(isChatTheActiveTab) {
+				this.forceTabsContentStyleWhenChatTabIsActive(isChatTheActiveTab)
+			},
+		},
+	},
+
+	beforeMount() {
+		this.$store.dispatch('setCurrentUser', getCurrentUser())
+	},
+
+	methods: {
+		async joinConversation() {
+			await this.getFileConversation()
+
+			joinConversation(this.token)
+		},
+
+		leaveConversation() {
+			leaveConversation(this.token)
+
+			this.$store.dispatch('updateToken', null)
+			this.$store.dispatch('updateFileIdForToken', null)
+		},
+
+		async getFileConversation() {
+			// Clear previous requests if there's one pending
+			this.cancelGetFileConversation('canceled')
+			// Get a new cancelable request function and cancel function pair
+			const { request, cancel } = CancelableRequest(getFileConversation)
+			// Assign the new cancel function to our data value
+			this.cancelGetFileConversation = cancel
+			// Make the request
+			try {
+				const response = await request({ fileId: this.fileId })
+				this.$store.dispatch('updateToken', response.data.ocs.data.token)
+				this.$store.dispatch('updateFileIdForToken', this.fileId)
+			} catch (exception) {
+				console.debug(exception)
```

Comment:
Don't print 2 messages ;) Put it in the else branch instead

Reply:
Ok.